mozdownload/treeherder.py

```diff
@@ -82,11 +82,11 @@ def query_builds_by_revision(self, revision, job_type_name='Build', debug_build=
 
             # Set filters to speed-up querying jobs
             kwargs = {
-                'platform': self.platform,
                 'option_collection_hash': option_hash,
                 'job_type_name': job_type_name,
                 'exclusion_profile': False,
             }
+            kwargs.update(self.get_treeherder_platform(self.platform))
```

Comment:
I think it's possible for the result of `self.get_treeherder_platform(self.platform)` to be just `self.platform`, which is a string, right? ...in which case the `update` call would fail, no?